project/ws/app/src/lib/routes/home/routes/users-view/users-view.component.ts

```diff
@@ -82,159 +84,271 @@ export class UsersViewComponent implements OnInit {
+    this.currentFilter = filter
+    this.pageIndex = 0
+    this.currentOffset = 0
+    this.limit = 20
+    this.searchQuery = ''
+    this.filterData(this.searchQuery)
+  }
+
+  get dataForTable() {
+    switch (this.currentFilter) {
+      case 'active':
+        return this.activeUsersData
+      case 'inactive':
+        return this.inactiveUsersData
+      default:
+        return []
+    }
+  }
+
+  filterData(query: string) {
+    let index = 0
+    if (this.currentFilter === 'active') {
+      index = 1
+      const data = { index, label: this.currentFilter }
+
+      this.raiseTabTelemetry(this.currentFilter, data)
+      this.activeUsers(query)
+    } else if (this.currentFilter === 'inactive') {
+      index = 2
+      const data = { index, label: this.currentFilter }
 
+      this.raiseTabTelemetry(this.currentFilter, data)
+      this.inActiveUsers(query)
+    }
+  }
+
+  activeUsers(query: string) {
+    this.loaderService.changeLoad.next(true)
+    const activeUsersData: any[] = []
+    const status = this.currentFilter === 'active' ? 1 : 0
+    this.currentOffset = this.limit * ((this.pageIndex + 1) - 1)
+    this.usersService.getAllKongUsersPaginated(this.rootOrgId, status, this.limit, this.currentOffset, query).subscribe((data: any) => {
+      this.userDataTotalCount = data.result.response.count
+      this.usersData = data.result.response
+      if (this.usersData && this.usersData.content && this.usersData.content.length > 0) {
+        _.filter(this.usersData.content, { isDeleted: false }).forEach((user: any) => {
+          // tslint:disable-next-line
+          const org = { roles: _.get(_.first(_.filter(user.organisations, { organisationId: _.get(this.configSvc, 'unMappedUser.rootOrg.id') })), 'roles') }
+          activeUsersData.push({
+            fullname: user ? `${user.firstName}` : null,
+            active: !user.isDeleted,
+            email: user.personalDetails && user.personalDetails.primaryEmail ?
+              this.profileUtilSvc.emailTransform(user.personalDetails.primaryEmail) : this.profileUtilSvc.emailTransform(user.email),
+            roles: org.roles.toString().replace(',', ', '),
+            userId: user.userId,
+            role: org.roles || [],
+            blocked: user.blocked,
+          })
+        })
+      }
+      this.activeUsersData = activeUsersData
+      return this.activeUsersData
+    })
+  }
+
+  inActiveUsers(query: string) {
+    this.loaderService.changeLoad.next(true)
+    const inactiveUsersData: any[] = []
+    const status = this.currentFilter === 'active' ? 1 : 0
+    this.currentOffset = this.limit * ((this.pageIndex + 1) - 1)
+    this.usersService.getAllKongUsersPaginated(this.rootOrgId, status, this.limit, this.currentOffset, query).subscribe(
+      (data: any) => {
+        this.userDataTotalCount = data.result.response.count
+        this.usersData = data.result.response
+        if (this.usersData && this.usersData.content && this.usersData.content.length > 0) {
```

Comment:
check for this.usersData.content.length as well then this.usersData.content.length>0

Reply:
fixed this, and pushed